saleor/graphql/checkout/dataloaders.py

```diff
@@ -637,27 +637,17 @@ def batch_load(self, keys):
         return [checkout_metadata.get(checkout_id) for checkout_id in keys]
 
 
-class ChannelByCheckoutLineIDLoader(DataLoader):
-    context_key = "channel_by_checkout_line"
+class ChannelByCheckoutIDLoader(DataLoader):
+    context_key = "channel_by_checkout"
 
     def batch_load(self, keys):
-        def channel_by_lines(checkout_lines):
-            checkout_ids = [line.checkout_id for line in checkout_lines]
+        def with_checkouts(checkouts):
+            channel_ids = [
+                checkout.channel_id if checkout else None for checkout in checkouts
```

Comment:
Have you checked this when `checkout` is None?
Also if you will have 100 checkouts, your `channel_ids` will have 100 records. Then `ChannelByIdLoader` will return 100 instances of the channels where some of them could be the same. The records will be stored on `ChannelByCheckoutIDLoader` loader. 
Better way, would be making `channel_ids = set(channel_ids)` and additional `then` that would make:
```
return [channel_map.get(channel_id) for channel_id in keys]
```